### PR TITLE
Add skills for maxtext developers

### PR DESCRIPTION
**Pull request by hengtaoguo**
# Description

Add a dedicated agent skill folder for maxtext developers.

# Tests

Tested with codex agent. My single prompt is:

> @skills Onboard microsoft/Phi-4-mini-instruct to maxtext

Validation branch in [here](https://github.com/AI-Hypercomputer/maxtext/compare/main...hengtaoguo-skill-validate). The agent wrote codes and performed checkpoint conversion for both scanned/unscanned version. Decode and logit check test both pass.

```
# Convert
python -m maxtext.checkpoint_conversion.to_maxtext src/maxtext/configs/base.yml model_name=phi4-mini-instruct base_output_directory=/dev/shm/hengtaoguo_google_com/phi4-full-scanned use_multimodal=false scan_layers=false hf_access_token=xxx weight_dtype=bfloat16 hardware=cpu skip_jax_distributed_system=True checkpoint_storage_use_ocdbt=False checkpoint_storage_use_zarr3=False --eager_load_method=safetensors --lazy_load_tensors=false
INFO:absl:Conversion complete. Checkpoint saved to /dev/shm/hengtaoguo_google_com/phi4-full-scanned
INFO:absl:Overall Elapse: 2.05 min
INFO:absl:Peak Memory: 21.03 GB

# Logits
python -m tests.utils.forward_pass_logit_checker src/maxtext/configs/base.yml run_name=ht_test model_name=phi4-mini-instruct tokenizer_path=microsoft/Phi-4-mini-instruct load_parameters_path=/dev/shm/hengtaoguo_google_com/phi4-full-scanned/0/items base_output_directory=/tmp/phi4-logits-check scan_layers=true skip_jax_distributed_system=true per_device_batch_size=1 max_target_length=32 max_prefill_predict_length=4 ici_tensor_parallelism=4 dtype=float32 weight_dtype=float32 activations_in_float32=true matmul_precision=highest float32_logits=true float32_qk_product=true attention=dot_product --run_hf_model=True --hf_model_path=microsoft/Phi-4-mini-instruct --trust_remote_code=False --max_kl_div=0.025 --skip_first_token
Max KL divergence for a single token in the set:7.06e-05

# Decode
python -m maxtext.inference.decode src/maxtext/configs/base.yml run_name=vllm_decode_qwen3_vl model_name=phi4-mini-instruct tokenizer_path=microsoft/Phi-4-mini-instruct load_parameters_path=/dev/shm/hengtaoguo_google_com/phi4-full-scanned/0/items tokenizer_type=huggingface prompt=\'Paris\ is\ the\' max_prefill_predict_length=16 max_target_length=32 per_device_batch_size=1 ici_tensor_parallelism=4 scan_layers=true weight_dtype=bfloat16 hf_access_token=xxx
Input `Paris is the` -> ` capital and most populous city of France, with an estimated population of 2,148`
```

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run end-to-end tests tests and provided workload links above if applicable.
- [x] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents (toctree directive) as explained in [our documentation](https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files).

### CODE REVIEW

File: .agents/skills/model-bringup/references/jax_layers.md [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+   - Map them exactly to your Flax model parameter dictionary.
+
+### Concrete Unit Test Template:
+
+```python
+import jax
+import jax.numpy as jnp
+import numpy as np
+import torch
+from maxtext.src.maxtext.layers import decoders as maxtext_decoders
```
[the remark this answers is not in the document]
Comment (hengtaoguo) [reply .agents/skills/model-bringup/references/jax_layers.md]: Done

File: .agents/skills/model-bringup/references/hf_analysis.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+4. This mini HF model will be used in Phase 3 for fast `to_maxtext` conversion and in Phase 4 for both `forward_pass_logit_checker` and 16-token autoregressive decoding comparison.
+
+```python
+# Quick snippet to slice HF model to mini safetensors subset
+import os
+from safetensors.torch import load_file, save_file
+from transformers import AutoConfig, AutoTokenizer
+
+model_id = "{hf_model_id}"
+out_dir = f"/dev/shm/hf_mini/{model_name}_{N}layers"
```
[the remark this answers is not in the document]
Comment (hengtaoguo) [reply .agents/skills/model-bringup/references/hf_analysis.md:68]: Done

File: .agents/skills/model-bringup/references/logits_checker.md [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+---
+
+## 3. Run MaxText Forward Pass Logits Checker on TPU VM
+Execute the logits checker script directly from the MaxText root directory on your TPU VM:
+
+```bash
+export HF_HOME=/dev/shm/$USER
+
+python3 -m maxtext.tests.utils.forward_pass_logit_checker   --model_name={model_name}   --maxtext_model_path=/dev/shm/$USER/checkpoints/{model_name}_mini_orbax   --golden_logits_path=/dev/shm/$USER/golden_mini_logits.jsonl   --base_num_decoder_layers={N}
```
[the remark this answers is not in the document]
Comment (hengtaoguo) [reply .agents/skills/model-bringup/references/logits_checker.md]: Done

**Review by bvandermoon — COMMENTED** [review]
@entrpn can you please take a look as well?

File: .agents/skills/model-bringup/references/hf_analysis.md [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+This guide covers Phase 1 of the MaxText model bring-up workflow: identifying the model, inspecting its architecture/codebase, analyzing decoder patterns, and preparing a minimal safetensors subset for rapid iteration.
+
+## 1. Locate the HF Model ID
+- Identify the exact model name on HuggingFace Hub (e.g., `Qwen/Qwen2.5-7B`, `google/gemma-2-9b`, `THUDM/glm-4-9b-chat`).
+- The model ID is typically formatted as `{organization}/{model_name}`.
+
+## 2. Download & Inspect HF Source Code
+> [!IMPORTANT]
+> **Avoid Filling Up TPU VM local disk space**:
```
[the remark this answers is not in the document]
Comment (hengtaoguo) [reply .agents/skills/model-bringup/references/hf_analysis.md:11]: Done, added a comment to allow user specify the cache directory instead.

File: .agents/skills/model-bringup/references/hf_analysis.md [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+# 1. Save modified config
+config = AutoConfig.from_pretrained(model_id, trust_remote_code=True)
+config.num_hidden_layers = N  # minimal representative subset
+config.save_pretrained(out_dir)
+
+# 2. Save tokenizer
+tokenizer = AutoTokenizer.from_pretrained(model_id, trust_remote_code=True)
+tokenizer.save_pretrained(out_dir)
+
+# 3. Slice and save safetensors weights (layers 0..N-1, embed_tokens, norm, lm_head)
```
[the remark this answers is not in the document]
Comment (hengtaoguo) [reply .agents/skills/model-bringup/references/hf_analysis.md:81]: This may not be necessary, as the agent can auto-complete the code snippet. I will follow up with another PR adding these standalone tools, then update the SKILL to point directly at those tools.

File: .agents/skills/model-bringup/references/logits_checker.md [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+
+prompt = "I love to"
+inputs = tokenizer(prompt, return_tensors="pt")
+
+# 1. Golden Forward Pass Logits
+with torch.no_grad():
+  outputs = model(**inputs)
+  golden_logits = outputs.logits.detach().cpu().numpy().tolist()
+
+# Save to golden_mini_logits.jsonl for forward_pass_logit_checker
```
[the remark this answers is not in the document]
Comment (hengtaoguo) [reply .agents/skills/model-bringup/references/logits_checker.md:38]: Done

File: .agents/README.md [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+workflow instructions. Use lowercase kebab-case for skill names, such as
+`model-bringup`, and match the directory name to the frontmatter `name`.
+Add `references/`, `scripts/`, or `assets/` inside a skill only when it needs
+supporting documentation, executable helpers, or templates.
+
+## Using the skills
+
+- **Codex:** discovers repository skills in `.agents/skills/`.
+- **Claude Code:** use the explicit file prompt or personal installation below.
+- **GitHub Copilot:** supports `.agents/skills/` as a repository
```
[the remark this answers is not in the document]
Comment (hengtaoguo) [reply .agents/README.md:27]: Typo fixed.

File: .agents/skills/model-bringup/references/checkpoint_conversion.md [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+
+- **Avoid filling up VM local space**: Always export `HF_HOME=/dev/shm/$USER` before running the tool.
+- **Fast Iteration on Mini Subset First**: Before converting full multi-gigabyte models, ALWAYS test conversion against your local mini safetensors subset created in `/dev/shm/hf_mini/{model_name}_xlayers`. Set `base_num_decoder_layers=N` in the command. This saves massive amounts of time and memory.
+- **Dump checkpoint to shared memory**: Always set `base_output_directory=/dev/shm/$USER/checkpoints` (or a subdirectory within it) for saving converted checkpoints. If subsequent logits checks fail, immediately delete the converted checkpoint using `rm -rf` to free up shared memory space for the next iteration.
+
+### Convert Mini HuggingFace Subset to MaxText (to_maxtext)
+```bash
+export HF_HOME=/dev/shm/$USER
+
+python3 -m maxtext.checkpoint_conversion.to_maxtext   src/maxtext/configs/base.yml   model_name={model_name}   hf_model_path=/dev/shm/hf_mini/{model_name}_xlayers   base_num_decoder_layers={N}   base_output_directory=/dev/shm/$USER/checkpoints/{model_name}_mini_orbax   scan_layers=true   hardware=cpu   skip_jax_distributed_system=True
```
[the remark this answers is not in the document]
Comment (hengtaoguo) [reply .agents/skills/model-bringup/references/checkpoint_conversion.md]: Done

**Review by entrpn — APPROVED** [review]

**Review by bvandermoon — APPROVED** [review]
